### DISCUSSION TITLE
Affect variable

### DISCUSSION THREAD

**Discussion by pcmanprogrammeur** [cypress-io/cypress#18620] [Questions and Help]
Hi,

The first `cy.log(link)` return empty string. Do you know why please ?

```
context('Test', () => {
    let link = "";

    before(async () => {
        cy.visit(Cypress.env('BASE_URL'));
        cy.get('a').first().invoke('attr', 'href')
            .then(href => {
                link = href
            });
        cy.log(link)
    });

    it('Test 1', () => {
        cy.log(link)
    });

    it('Test 2', () => {
        cy.log(link)
    });
});
```

**Comment by marktnoonan** [comment]
Hi! Looks like your first `console.log` call is outside the `then`, so it runs before that `href` has been assigned to the `link` variable.

**Comment by bahmutov** [comment]
1: you cannot use async keyword and functions in cypress hooks and tests 
2: find a video why cy log prints null among my videos at https://cypress.tips/search
